pkg/uplinkdb/uplinkdb.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplinkdb
+
+import (
+	"context"
+	"crypto"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// DB stores bandwidth agreements.
+type DB interface {
+	// SavePublicKey adds a new bandwidth agreement.
+	SavePublicKey(context.Context, Agreement) error
+	// GetPublicKey gets the public key of uplink corresponding to serial number
+	GetPublicKey(ctx context.Context, nodeID []byte) (*Agreement, error)
+}
+
+// Server is an implementation of the pb.BandwidthServer interface
+type Server struct {
+	db     DB
+	pkey   crypto.PublicKey
+	logger *zap.Logger
+}
+
+// Agreement is a struct that contains a uplinks agreement info
+type Agreement struct {
+	ID        []byte // uplink id
+	PublicKey []byte // uplink public key
+	CreatedAt time.Time
+}
+
+// NewServer creates instance of Server
+func NewServer(db DB, logger *zap.Logger, pkey crypto.PublicKey) *Server {
```

Comment:
It won't be server. It should be similar to `irreparable`, just database interface